From acd6b191e55c26d5ae9f5e79af3d17e483d9374f Mon Sep 17 00:00:00 2001
From: Ramalingam C <ramalingam.c@intel.com>
Date: Thu, 18 Jan 2018 11:18:06 +0530
Subject: [PATCH 0536/1240] UPSTREAM: drm/i915: II stage HDCP auth for repeater
 only

Second stage of HDCP authentication required only if the HDCP
sink is a repeater.

This patch imposes above condition on second stage HDCP authentication.

Signed-off-by: Ramalingam C <ramalingam.c@intel.com>
Signed-off-by: Sean Paul <seanpaul@chromium.org>
Link: https://patchwork.freedesktop.org/patch/msgid/1516254488-4971-3-git-send-email-ramalingam.c@intel.com
(cherry picked from commit a5f7869b406c4b002e4b68ccdfba69177de8668e)

Change-Id: I24e57740b90ffee0221fb075c71d1f44fe61d1d8
---
 drivers/gpu/drm/i915/intel_hdcp.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_hdcp.c b/drivers/gpu/drm/i915/intel_hdcp.c
index e49ca7f..c8e611a 100644
--- a/drivers/gpu/drm/i915/intel_hdcp.c
+++ b/drivers/gpu/drm/i915/intel_hdcp.c
@@ -493,7 +493,10 @@ static int intel_hdcp_auth(struct intel_digital_port *intel_dig_port,
 	 * on those as well.
 	 */
 
-	return intel_hdcp_auth_downstream(intel_dig_port, shim);
+	if (repeater_present)
+		return intel_hdcp_auth_downstream(intel_dig_port, shim);
+
+	return 0;
 }
 
 static
-- 
2.7.4

